safe_control_gym/experiment.py

```diff
@@ -1,148 +1,238 @@
-        We do not standarize on the training env as input to this method since different controllers use it differently. 
+    def __init__(self, 
+                 env, 
+                 ctrl, 
+                 safety_filter = None,
+                ):
+        """Creates a generic experiment class to run evaluations and collect standard metrics.
+
+        Args:
+            env (gym.Env): the environment for the task.
+            ctrl (BaseController): the controller for the task
+            safety_filter (BaseSafetyFilter): the safety filter to filter the controller
         """
-        ctrl.reset()
-        if config.restore:
-            ctrl.load(os.path.join(config.restore, "model_latest.pt"))
-        ctrl.learn()
-        print("Training done.")
+        
+        self.env = env
+        if not is_wrapped(self.env, RecordDataWrapper):
+            self.env = RecordDataWrapper(self.env)
+        self.ctrl = ctrl
+        self.safety_filter = safety_filter
     
-    def run_evaluation(self, config, ctrl, env, n_episodes=10, **kwargs):
-        """Evaluate a trained controller.
-
-        Note this evaluation uses the ctrl's own control frequency.
-        Optionally, use the ctrl's own `run()` method to evaluate instead of collecting full traj data.
+    def run_evaluation(self, n_episodes=10, log_freq=None, n_steps=None, **kwargs):
+        """Evaluate a trained controller. 
+        
+        Args:
+            n_episodes (int): number of runs to execute
+            n_steps (int): the number of steps to collect in total
+            log_freq (int): the frequency with which to log information
+        
+        Returns:
+            trajs_data (defaultdict(list)): the raw data from the executed runs 
+            metrics (dict): the metrics calculated from the raw data
         """
-        ctrl.reset()
-        if config.restore:
-            ctrl.load(os.path.join(config.restore, "model_latest.pt"))
-        trajs_data = self.collect_trajs(ctrl, env, n_trajs=n_episodes)
-        results = self.compute_metrics(trajs_data)
+
+        self.reset()
+        trajs_data = self._execute_evaluations(log_freq=log_freq, n_episodes=n_episodes, n_steps=n_steps, **kwargs)
+        metrics = self.compute_metrics(trajs_data)
+        
         # terminal printouts
-        for metric_key, metric_val in results.items():
+        for metric_key, metric_val in metrics.items():
             print("{}: {:.3f}".format(colored(metric_key,"yellow"), metric_val))
         print("Evaluation done.")
-        return results
+        return trajs_data, metrics
     
-    def collect_trajs(self, ctrl, env, log_freq=None, n_trajs=None, n_steps=None, **kwargs):
-        """"""
-        if not is_wrapped(env, RecordDataWrapper):
-            env = RecordDataWrapper(env)
+    def _execute_evaluations(self, n_episodes=10, n_steps=None, log_freq=None, **kwargs):
```

Comment:
with the two error checks, the `n_episodes` arg default should be `None` ?

Reply:
You're totally right, I'll fix that now